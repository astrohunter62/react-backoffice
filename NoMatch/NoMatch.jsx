import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import Typography from 'material-ui/Typography'
import withStyles from 'material-ui/styles/withStyles'

const styles = theme => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    maxWidth: '60rem',
    paddingLeft: theme.spacing.unit * 3,
    paddingRight: theme.spacing.unit * 3,
    marginLeft: 'auto',
    marginRight: 'auto',
    height: 'calc(100vh - 104px)',
  },
})

const NoMatch = ({ classes }) => (
  <div className={classes.root}>
    <div>
      <Typography type="display3">
        404! Sorry, not found.
      </Typography>
      <Typography type="display1" paragraph>
        This URL does not exist, sorry. Please start over from
        the <Link href="/">Dashboard</Link>.
      </Typography>
    </div>
  </div>
)

NoMatch.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(NoMatch)
