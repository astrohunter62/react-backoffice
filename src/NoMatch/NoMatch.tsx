import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import { Typography, withStyles } from "@material-ui/core";

const styles = (theme: any) => ({
  root: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    maxWidth: "60rem",
    paddingLeft: theme.spacing(3),
    paddingRight: theme.spacing(3),
    marginLeft: "auto",
    marginRight: "auto",
    height: "calc(100vh - 104px)",
  },
});

type NoMatchProps = {
  title?: string;
  description?: React.ReactNode;
  classes: {
    [key: string]: string;
  };
};

const NoMatch: React.SFC<NoMatchProps> = ({ title, description, classes }) => (
  <div className={classes.root}>
    <div>
      <Typography variant="h2">{title}</Typography>
      <Typography variant="h4" paragraph>
        {description}
      </Typography>
    </div>
  </div>
);

NoMatch.defaultProps = {
  title: "404! Sorry, not found.",
  description: (
    <Fragment>
      This URL does not exist, sorry. Please start over from the{" "}
      <Link href="/" to="/">
        Dashboard
      </Link>
      .
    </Fragment>
  ),
};

export default withStyles(styles)(NoMatch);
