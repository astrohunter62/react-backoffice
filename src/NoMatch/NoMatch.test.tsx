import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";

import NoMatch from ".";

it("renders correctly", () => {
  const tree = shallow(
    <Router>
      <NoMatch title="Title" />
    </Router>,
  );

  expect(tree).toMatchSnapshot();
});

it("renders correctly with description", () => {
  const tree = shallow(<NoMatch title="Title" description={<p>Desc</p>} />);

  expect(tree).toMatchSnapshot();
});
