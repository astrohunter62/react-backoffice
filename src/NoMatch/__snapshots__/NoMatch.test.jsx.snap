// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <WithStyles(NoMatch)
    title="Title"
  />
</Router>
`;

exports[`renders correctly with description 1`] = `
<NoMatch
  classes={
    Object {
      "root": "NoMatch-root-1",
    }
  }
  description={
    <p>
      Desc
    </p>
  }
  title="Title"
/>
`;
